modules/pytket-honeywell/pytket/extensions/honeywell/backends/honeywell.py

```diff
@@ -363,10 +364,13 @@ def process_circuits(
 
             if final_index > 0:
                 # only set batch fields if more than one job submitted
-                body["batch-exec"] = batch_exec
-                if i == final_index:
-                    # flag to signal end of batch
-                    body["batch-end"] = True
+                if (self._device_name != DEVICE_FAMILY) or ("max_batch_cost" in kwargs):
```

Comment:
The only suggestion I would make is `and`ing this condition with the `if final_index >0` so there is a single decision point as to whether to batch or not.

Reply:
Yeah that makes sense, will do!